src/transport/RendezvousSession.cpp

```diff
@@ -300,20 +301,24 @@ CHIP_ERROR RendezvousSession::HandlePairingMessage(PacketBuffer * msgBuf)
     return err;
 }
 
-CHIP_ERROR RendezvousSession::HandleSecureMessage(PacketBuffer * msgBuf)
+CHIP_ERROR RendezvousSession::HandleSecureMessage(PacketBuffer * msgIn)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
     PacketHeader packetHeader;
     PayloadHeader payloadHeader;
     MessageAuthenticationCode mac;
-    uint16_t headerSize            = 0;
-    uint8_t * data                 = nullptr;
-    uint8_t * plainText            = nullptr;
-    uint16_t len                   = 0;
-    uint16_t decodedSize           = 0;
-    uint16_t taglen                = 0;
-    uint16_t payloadlen            = 0;
-    System::PacketBuffer * origMsg = nullptr;
+    uint16_t headerSize  = 0;
+    uint8_t * data       = nullptr;
+    uint8_t * plainText  = nullptr;
+    uint16_t len         = 0;
+    uint16_t decodedSize = 0;
+    uint16_t taglen      = 0;
+    uint16_t payloadlen  = 0;
+
+    System::AutoFreePacketBuffer msgBuf(msgIn);
+    System::AutoFreePacketBuffer origMsg;
+
+    VerifyOrExit(!msgBuf.IsNull(), err = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Because msgBuf->Start() is used. I believe the missing check was an oversight (otherwise there should be a comment stating 'we know msgBuf is not null because ...'.

I really wish we had non_null annotations/data types.

Reply:
That's fair.  I'm fine with having the null-check, but eventually we may want to just declare that a PacketBuffer argument had better never be null, so we don't sprinkle null-checks everywhere.